denlillemand/week3/ass4/app/mainlist.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import NamedList from './namedlist.jsx';
+import NamedListForm from './namedlistform.jsx';
+
+export default class MainList extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {};
+        this.swapMessage = this.swapMessage.bind(this);
+        this.archiveMessage = this.archiveMessage.bind(this);
+        this.onSubmitNamedList = this.onSubmitNamedList.bind(this);
+        this.deleteMessage = this.deleteMessage.bind(this);
+        this.onSubmitMessage = this.onSubmitMessage.bind(this);
+    }
+
+    archiveMessage(id, currentList, isArchived) {
+        var mutatedArray = this.state[currentList].map(function (message) {
+            if (message.id === id) {
+                var clonedMessage = Object.assign({}, message);
+                clonedMessage.archived = !isArchived;
+                return clonedMessage;
+            }
+
+            return message;
+        });
+
+        this.setState({
+            [currentList]: mutatedArray,
+        });
+    }
+
+    swapMessage(messageId, destinationList, currentList) {
+        let messageIndex = -1;
+        this.state[currentList].forEach(function (message, index, array) {
+            if (message.id === messageId) {
+                messageIndex = index;
+            }
+        });
```

Comment:
you can use `findIndex` here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex